src/core/lib/slice/slice.cc

```diff
@@ -215,7 +215,9 @@ grpc_slice grpc_slice_from_copied_buffer(const char* source, size_t length) {
     slice = grpc_core::UnmanagedMemorySlice(
         length, grpc_core::UnmanagedMemorySlice::ForceHeapAllocation());
   }
-  memcpy(GRPC_SLICE_START_PTR(slice), source, length);
+  if (length > 0) {
```

Comment:
I didn't know memcpy with a size of 0 is ubsan?

Reply:
Passing a null pointer to memcpy is considered as an UB.
(https://en.cppreference.com/w/c/string/byte/memcpy)

Reply:
please change it to `source != nullptr` to make it clear then

Reply:
This is intentionally done because this should catch the error case of where it gets `source = nullptr` and `length > 0`.